### ISSUE TITLE
unable to install azure cli 2.0 in centos 6.9

### ISSUE THREAD

**Issue by saimsh** [Azure/azure-cli#4634]
### Description
**unable to successfully install Azure cli 2.0 in centos6.9 following the steps in the official link : 
https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest**

I am from azure linux escalation support team and I was able to reproduce one of my customer's issue:
---

### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here: Using yum 

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here: 2.0 

**OS Version:** What OS and version are you using?  
Answer here: Centos 6.9 (roguewave from azure gallery)

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here: Bash 

ERRORS:
[root@sscentos69 ~]# rpm --import https://packages.microsoft.com/keys/microsoft.asc
[root@sscentos69 ~]# sh -c 'echo -e "[azure-cli]\nname=Azure CLI\nbaseurl=https://packages.microsoft.com/yumrepos/azure-cli\nenabled=1\ngpgcheck=1\ngpgkey=https://packages.microsoft.com/keys/microsoft.asc" > /etc/yum.repos.d/azure-cli.repo'
[root@sscentos69 ~]# yum check-update
....
... ( outputs abstracted here)
[root@sscentos69 ~]# yum install azure-cli
Loaded plugins: fastestmirror, security
Setting up Install Process
Determining fastest mirrors
Resolving Dependencies
--> Running transaction check
---> Package azure-cli.x86_64 0:2.0.18-1.el7 will be installed
--> Processing Dependency: libpython2.7.so.1.0()(64bit) for package: azure-cli-2.0.18-1.el7.x86_64
--> Finished Dependency Resolution
Error: Package: azure-cli-2.0.18-1.el7.x86_64 (azure-cli)
           Requires: libpython2.7.so.1.0()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
[root@sscentos69 ~]# python -V
Python 2.6.6
next, I attempted to update python and other dependencies as per the prereqs:
[root@sscentos69 ~]# yum install -y gcc python libffi-devel python-devel openssl-devel
......
same errors persisted 
Finally , I updated the python version using the following steps:
[root@sscentos69 ~]# yum install centos-release-scl
....
[root@sscentos69 ~]# yum install python27
.....
[root@sscentos69 ~]# python -V
Python 2.6.6
[root@sscentos69 ~]# scl enable python27 bash
[root@sscentos69 ~]# python -V
Python 2.7.13
[root@sscentos69 ~]# yum install azure-cli

the installation was successful after this point. however, the following errors persisted:
[root@sscentos69 ~]# az
Could not import runpy module
[root@sscentos69 ~]#

my first ask would be, is azure cli 2.0 supported for centos versions < 7.0 ? 
If so, what steps am I missing here?

thanks,
saim

**Comment by derekbekoe** [comment]
Duplicate of https://github.com/Azure/azure-cli/issues/4581.
The RPM package supports RHEL/CentOS 7 as indicated by the `el7`.

CentOS 6 has Python 2.6 and the CLI doesn't support that.

**Comment by walterheck** [comment]
@derekbekoe that's too easy and not true either: as @saimsh clearly demonstrates in his post python 2.7 is a RedHat software collection for RHEL 6 and fully supported.
from https://access.redhat.com/documentation/en-us/red_hat_software_collections/1/html/1.2_release_notes/chap-rhscl: 

> all Red Hat Software Collections components are fully supported under Red Hat Enterprise Linux Subscription Level Agreements, are functionally complete, and are intended for production use.

**Comment by agimenez** [comment]
Yeah, can we reopen this?

**Comment by dschinnerl** [comment]
Hi,

I'm a bit late to the game but if since I got here by a big search engine...

I solved it via:

```
#!/usr/bin/env bash
#/usr/lib64/az/bin/python -Esm azure.cli "$@"

export LD_LIBRARY_PATH=/opt/rh/python27/root/usr/lib64
export PYTHONPATH=/usr/lib64/az/lib/python2.7:/usr/lib64/az/lib/python2.7/site-packages

/opt/rh/python27/root/usr/bin/python2.7 -sm azure.cli "$@"

# -E     : ignore PYTHON* environment variables (such as PYTHONPATH)
# -s     : don't add user site directory to sys.path; also PYTHONNOUSERSITE
```

```
[hudson@xxxxxx ~]$ cat /etc/redhat-release ; /usr/bin/az
CentOS release 6.10 (Final)

     /\
    /  \    _____   _ _  ___ _
   / /\ \  |_  / | | | \'__/ _\
  / ____ \  / /| |_| | | |  __/
 /_/    \_\/___|\__,_|_|  \___|


Welcome to the cool new Azure CLI!
```

Maybe it helps someone...

**Comment by Moeser** [comment]
@dschinnerl you sir, are a life saver.  For others running into this, I didn't need LD_LIBRARY_PATH and the python2.7 bin was located in a different place, but other than that, my /usr/bin/az looks similar to the above and works now.
